pkg/cloud/aws/services/ec2/subnets.go

```diff
@@ -255,21 +288,23 @@ func (s *Service) deleteSubnet(id string) error {
 }
 
 func (s *Service) getSubnetTagParams(id string, public bool) tags.BuildParams {
-	var name strings.Builder
-
-	name.WriteString(s.scope.Name())
-	name.WriteString("-subnet-")
+	var role string
 	if public {
-		name.WriteString("public")
+		role = tags.ValuePublicRole
 	} else {
-		name.WriteString("private")
+		role = tags.ValuePrivateRole
 	}
 
+	var name strings.Builder
```

Comment:
`fmt.Sprtintf` would be easier to read. I believe string builder was used to avoid a var declaration, but it ended up using a var declaration anyway. 

The more go-like pattern I'd expect here is

```go
role := tags.ValuePrivateRole
if public {
    role = tags.ValuePublicRole
}
name := fmt.Sprintf("%s-subnet-%s", s.scope.Name(), role)
```

This is pretty minor though.